specification/eventgrid/Azure.Messaging.EventGrid/main.tsp

```diff
@@ -0,0 +1,243 @@
+import "@typespec/http";
+import "@typespec/rest";
+import "@typespec/versioning";
+import "@azure-tools/typespec-autorest";
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-client-generator-core";
+
+enum ServiceApiVersions {
+  v2023_06_01_preview: "2023-06-01-preview"
+}
+
+@useAuth(
+  ApiKeyAuth<ApiKeyLocation.header, "SharedAccessKey"> | OAuth2Auth<[{
+    type: OAuth2FlowType.implicit,
+    authorizationUrl: "https://login.microsoftonline.com/common/oauth2/v2.0/authorize",
+    scopes: ["https://eventgrid.azure.net/.default"],
+  }]>
+)
+
+@service({
+  title: "Azure.Messaging.EventGridMessagingClient",
+  version: "2023-06-01-preview"
+})
+
+//
+// Supported operations.
+//
+// POST https://{namespaceName}.{region}.eventgrid.azure.net/topics/{topicName}:publish?api-version={apiVersion}}
+// POST https://{namespaceName}.{region}.eventgrid.azure.net/topics/{topicName}/eventsubscriptions/{eventSubscriptionName}:receive?api-Version={apiVersion}&timeout=60&maxEvents={maxEvents}
+// POST https://{namespaceName}.{region}.eventgrid.azure.net/topics/{topicName}/eventsubscriptions/{eventSubscriptionName}:acknowledge?api-Version={apiVersion}
+// POST https://{namespaceName}.{region}.eventgrid.azure.net/topics/{topicName}/eventsubscriptions/{eventSubscriptionName}:release?api-version={apiVersion}
+//
+
+@server(
+  "{endpoint}",
+  "The host name of the topic",
+  {
+    @doc("The host name of the namespace, e.g. namespaceName1.westus-1.eventgrid.azure.net")
+    endpoint: url,
+  }
+)
+
+@doc("Azure Messaging EventGrid Client")
+@versioned(ServiceApiVersions)
+namespace Azure.Messaging.EventGridMessaging {
+  using TypeSpec.Http;
+  using TypeSpec.Rest;
+  using TypeSpec.Versioning;
+  using Azure.Core;
+  using Azure.Core.Foundations;
+  using Azure.ClientGenerator.Core;
+
+  @doc("Properties of an event published to an Azure Messaging EventGrid Namespace topic using the CloudEvent 1.0 Schema.")
+  @internal
+  model CloudEventEvent {
+    @doc("An identifier for the event. The combination of id and source must be unique for each distinct event.")
+    id: string;
+
+    @doc("Identifies the context in which an event happened. The combination of id and source must be unique for each distinct event.")
+    source: string;
+
+    @doc("Event data specific to the event type.")
+    data?: object;
```

Comment:
Should this be changed to `unknown`?

Reply:
https://github.com/Azure/azure-rest-api-specs/pull/23539/files